addons/payment_authorize/models/payment_transaction.py

```diff
@@ -95,18 +95,37 @@ def _send_refund_request(self, amount_to_refund=None, create_refund_transaction=
                 create_refund_transaction=create_refund_transaction,
             )
 
-        refund_tx = super()._send_refund_request(
-            amount_to_refund=amount_to_refund, create_refund_transaction=False
-        )
+        # Only full refund is possible.
 
         authorize_API = AuthorizeAPI(self.acquirer_id)
-        rounded_amount = round(self.amount, self.currency_id.decimal_places)
-        res_content = authorize_API.refund(self.acquirer_reference, rounded_amount)
+        tx_details = authorize_API.get_transaction_details(self.acquirer_reference)
+
+        if tx_details and tx_details.get('err_code'):
+            res_content = {
+                'x_response_code': AuthorizeAPI.AUTH_ERROR_STATUS,
+                'x_response_reason_text': tx_details.get('err_msg')
+            }
+
+            refund_tx = super()._send_refund_request(
+                amount_to_refund=amount_to_refund, create_refund_transaction=False
+            )
+        elif tx_details.get('transaction', {}).get('transactionStatus') in ['capturedPendingSettlement', 'authorizedPendingCapture']:
+            refund_tx = super()._send_refund_request(
+                amount_to_refund=self.amount, create_refund_transaction=True
+            )
+            res_content = authorize_API.void(self.acquirer_reference)
+        else:
+            refund_tx = super()._send_refund_request(
+                amount_to_refund=amount_to_refund, create_refund_transaction=True
+            )
+
+            # continue with the usual refund
+            # We continue to use the full amount below
+            rounded_amount = round(amount_to_refund, self.currency_id.decimal_places)
+            res_content = authorize_API.refund(self.acquirer_reference, rounded_amount)
+
         _logger.info("refund request response:\n%s", pprint.pformat(res_content))
-        # As the API has no redirection flow, we always know the reference of the transaction.
-        # Still, we prefer to simulate the matching of the transaction by crafting dummy feedback
-        # data in order to go through the centralized `_handle_feedback_data` method.
-        feedback_data = {'reference': self.reference, 'response': res_content}
+        feedback_data = {'reference': refund_tx.reference, 'response': res_content}
```

Comment:
In a validation case, the refund_tx will be self.env['payment.transaction'], not the validation tx reference ;)

```suggestion
        feedback_data = {'reference': refund_tx.reference or self.reference, 'response': res_content}
```

Reply:
done :)